go/chat/storage/storage.go

```diff
@@ -436,3 +438,84 @@ func (s *Storage) FetchMessages(ctx context.Context, convID chat1.ConversationID
 
 	return res, nil
 }
+
+type BodyHashChecker func(bodyHash chat1.Hash, uniqueMsgID chat1.MessageID, uniqueConvID chat1.ConversationID) error
+type PrevChecker func(msgID chat1.MessageID, convID chat1.ConversationID, uniqueHeaderHash chat1.Hash) error
+
+// These are globally unique. They don't include the UID.
+func makeBodyHashIndexKey(bodyHash chat1.Hash) libkb.DbKey {
+	return libkb.DbKey{
```

Comment:
It would be great if this includes a version number so we can bust it easily if we bungle something. Haven't read this part too closely, could've missed that.